src/main/java/de/jtheb/fh/lagerverwaltung/entities/Shelf.java

```diff
@@ -32,11 +34,16 @@ public boolean isFull() {
 
     public Compartment findFittingCompartment(Item item) {
         for (Compartment compartment : this.compartments) {
-            if (compartment.itemFits(item)) {
+            if (compartment.itemFitsWithArticleNr(item)) {
+                return compartment;
+            }
+        }
+        for (Compartment compartment : compartments) {
+            if (compartment.getArticleNr().equals(null)) {
                 return compartment;
             }
         }
-        throw new UnsupportedOperationException("not implemented"); //TODO ERROR Function/Shelffull
+        throw new UnsupportedOperationException("Error - compartements full"); //TODO Correct ERROR Function/Shelffull
```

Comment:
Find a better checked Exception (s.th. that is not a RuntimeException), this is a RuntimeException. Write your own exception, consult me if you need help.